libs/ledgerjs/packages/hw-app-eth/src/modules/EIP712/index.ts

```diff
@@ -218,6 +241,109 @@ const EIP712SendStructImplem = async (
   return Promise.resolve();
 };
 
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "activate"
+): Promise<Buffer>;
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "contractName",
+  data: FilteringInfoContractName
+): Promise<Buffer>;
+async function sendFilteringInfo(
+  transport: Transport,
+  type: "showField",
+  data: FilteringInfoShowField
+): Promise<Buffer>;
```

Comment:
Why these 3 definitions?

Reply:
Yup, function overloading to make sure TS can recognize that:
- if `type === "activate"` -> `data = undefined`
- if `type === "contractName"` -> `data` as to be of type `FilteringInfoContractName`
- if `type === "showField"` -> `data` as to be of type `FilteringInfoShowField `

Anything else should be flagged as wrong by TS.